lib/qless/middleware/timeout.rb

```diff
@@ -15,29 +16,45 @@ module Middleware
     # or a variable timeout based on the individual TTLs of each job
     # (using something like `extend Qless::Middleware::Timeout.new { |job| job.ttl * 1.1 }`).
     class Timeout < Module
-      def initialize
+      def initialize(opts = {})
+        timeout_class = opts.fetch(:timeout_class, ::Timeout)
+        kernel_class = opts.fetch(:kernel_class, Kernel)
         module_eval do
           define_method :around_perform do |job|
-            timeout_value = yield job
-            if timeout_value <= 0
-              raise InvalidTimeoutError, "Timeout must be positive, but was #{timeout_value}"
-            end
+            timeout_seconds = yield job
+            if timeout_seconds.nil?
+              super(job)
```

Comment:
You can turn this `if/else` into an early return:

``` ruby
return super(job) if timeout_seconds.nil?

# the else logic goes here w/o a need to be nested under an `else`
```

I find that to be a bit cleaner.